voicevox_engine/resource_manager.py

```diff
@@ -0,0 +1,59 @@
+"""
+URLから取得可能なリソースの管理
+
+アップデートでリソースの変更が反映されるようにURLはハッシュ値から生成する
+"""
+
+import base64
+import json
+from hashlib import sha256
+from pathlib import Path
+from typing import Literal
+
+
+class ResourceManagerError(Exception):
+    def __init__(self, message: str):
+        self.message = message
+
+
+def b64encode_str(s: bytes) -> str:
+    return base64.b64encode(s).decode("utf-8")
+
+
+class ResourceManager:
+    def __init__(self, create_filemap_if_not_exist: bool) -> None:
+        self._create_filemap_if_not_exist = create_filemap_if_not_exist
+        self._path_to_hash: dict[Path, str] = {}
+        self._hash_to_path: dict[str, Path] = {}
+
+    def register_dir(self, resource_dir: Path) -> None:
+        filemap_json = resource_dir / "filemap.json"
+        if filemap_json.exists():
+            data: dict[str, str] = json.loads(filemap_json.read_bytes())
+            self._path_to_hash |= {resource_dir / k: v for k, v in data.items()}
+        else:
+            if self._create_filemap_if_not_exist:
+                self._path_to_hash |= {
+                    i: sha256(i.read_bytes()).digest().hex()
+                    for i in resource_dir.glob("**/*")
```

Comment:
`**/*`がなにかわからないかもなので（？）、rglob使うとわかりやすい･･･というか意図を追えるかも。
```suggestion
                    for i in resource_dir.rglob("*")
```